app/models/interface.rb

```diff
@@ -0,0 +1,30 @@
+class Interface < ApplicationRecord
+  enum interface_type: { discord_guild: 0, web: 1, api: 2 }
+
+  before_save :ensure_single_instance, if: -> { api? || web? }
+
+  has_many :messages, dependent: :destroy
+
+  validates :interface_type, presence: true
+  validates :external_id, presence: true, if: :requires_external_id?
+  validates :external_id, uniqueness: { scope: :interface_type }, if: :requires_external_id?
+
```

Comment:
Sorry, I think we do need to have a method like the ones I asked you to delete 😬, to get the api or web interfaces:

```rb
class << self
  def api
    find_by(source: :api)
  end

  def web
    find_by(source: :web)
  end
end
```

Reply:
Please create a rake task to create the `web` and `source` interfaces. Or a rails migration

Reply:
ok, done!